lib/resourceapply/machineconfig.go

```diff
@@ -80,21 +81,32 @@ func ApplyMachineConfigNode(client mcfgclientalphav1.MachineConfigNodesGetter, r
 
 // ApplyControllerConfig applies the required machineconfig to the cluster.
 func ApplyControllerConfig(client mcfgclientv1.ControllerConfigsGetter, required *mcfgv1.ControllerConfig) (*mcfgv1.ControllerConfig, bool, error) {
+	klog.Info("Getting existing ControllerConfig with name: ", required.GetName())
 	existing, err := client.ControllerConfigs().Get(context.TODO(), required.GetName(), metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
+		klog.Infof("ControllerConfig not found, creating new one")
 		actual, err := client.ControllerConfigs().Create(context.TODO(), required, metav1.CreateOptions{})
+		if err != nil {
+			klog.Errorf("Failed to create ControllerConfig: %v", err)
+		}
 		return actual, true, err
 	}
 	if err != nil {
+		klog.Errorf("Error fetching ControllerConfig: %v", err)
 		return nil, false, err
 	}
 
 	modified := resourcemerge.BoolPtr(false)
 	mcoResourceMerge.EnsureControllerConfig(modified, existing, *required)
 	if !*modified {
+		klog.Info("No updates required for the ControllerConfig")
 		return existing, false, nil
 	}
 
+	klog.Infof("Updating existing ControllerConfig")
```

Comment:
suggestion (non-blocking): We may not want to log this every time because it would lead to a lot of additional logs being written. Instead, we may want to put this behind a `klog.V(4)` instead so that it's only enabled whenever `-v4` is given at startup.